src/pages/Main/ClassList.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { FaHeart } from 'react-icons/fa';
+import { FaRegHeart } from 'react-icons/fa';
+
+function ClassList({ id, thumbnail_image, creator, name, like_count, price }) {
+  const [isLike, setIsLike] = useState(false);
+  const [countLike, setCountLike] = useState(like_count);
+
+  const handleLike = () => {
+    if (isLike) {
+      fetch('api주소', {
+        method: 'POST',
+        body: JSON.stringify({
+          classId: id,
+          like_count: countLike,
+        }),
+      });
+      setIsLike(!isLike);
+    } else {
+      fetch('api주소', {
+        method: 'DELETE',
+      });
+      setIsLike(!isLike);
+    }
+  };
+
+  useEffect(() => {
+    if (isLike) {
+      setCountLike(countLike + 1);
+    } else {
+      setCountLike(countLike - 1);
+    }
```

Comment:
```suggestion
        isLike ? setCountLike(prev => prev + 1) : setCountLike(prev => prev - 1);
```
이 부분은 삼항 연산자로 구성할 수 있겠습니다!